modules/hostap/src/supp_main.c

```diff
@@ -415,7 +415,8 @@ static void iface_cb(struct net_if *iface, void *user_data)
 		return;
 	}
 
-	if (wifi_nm_get_type_iface(iface) != (1 << WIFI_TYPE_STA)) {
+	if ((wifi_nm_get_type_iface(iface) == 0) ||
+	    (wifi_nm_get_type_iface(iface) != (1 << WIFI_TYPE_STA))) {
```

Comment:
Instead of duplicating this multiple times, why not make an inline function that hides the checks, so something like
```
static inline bool wifi_is_station(struct net_if *iface)
{
	if ((wifi_nm_get_type_iface(iface) == 0) ||
	    (wifi_nm_get_type_iface(iface) != (1 << WIFI_TYPE_STA))) {
		return true;
	}

	return false;
}
```

Reply:
The function `net_if_get_wifi_sta` itself is a wrapper for `is_sta`, so, just inlined it. But I can add this to the `wifi_nm` API and use that here.

Reply:
Re-thinking, want to keep the unset type separate from explicit STA/AP in the Wi-FI NM APIs, so, how about this?

```
diff --git a/subsys/net/ip/net_if.c b/subsys/net/ip/net_if.c
index fc4945174c0..9154c149ad9 100644
--- a/subsys/net/ip/net_if.c
+++ b/subsys/net/ip/net_if.c
@@ -5689,14 +5689,14 @@ struct net_if *net_if_get_wifi_sta(void)
        STRUCT_SECTION_FOREACH(net_if, iface) {
                if (net_if_is_wifi(iface)
 #ifdef CONFIG_WIFI_NM
-                       && ((wifi_nm_get_type_iface(iface) == 0) ||
-                           (wifi_nm_get_type_iface(iface) == (1 << WIFI_TYPE_STA)))
+                       && wifi_nm_iface_is_sta(iface)
 #endif
                        ) {
                        return iface;
                }
        }
-       return NULL;
+
+       return net_if_get_first_wifi();
 }
```